### ISSUE TITLE
functions in alerting templates don't work as in Prometheus

### ISSUE THREAD

**Issue by fmbiete** [VictoriaMetrics/VictoriaMetrics#989]
**Describe the bug**
query, first, value functions in alerting templates don't work as they do in Prometheus

**To Reproduce**
Use this rule file
```
groups:
- name: test
  rules:

  - record: test_record
    expr: 1

  - alert: test_alert
    expr: up == 1
    for: 5m
    labels:
      dynamic: '{{ $x := query "test_record" | first | value }}{{ if eq 1.0 $x }}one{{ else }}unknown{{ end }}'
```

Validate the file; it will return an error, but it's valid for Prometheus

```
2021-01-04T13:42:18.257Z	fatal	/home/builder/rpm/BUILD/VictoriaMetrics-pmm-6401-v1.50.2/app/vmalert/main.go:67	errors(1): invalid group "test" in file "/srv/prometheus/rules/example.yml": invalid labels for rule "test"."tes_alert": errors(1): key "dynamic", template "{{ $x := query \"test_record\" | first | value }}{{ if eq 1.0 $x }}one{{ else }}unknown{{ end }}": error evaluating annotation template: template: :1:97: executing "" at <first>: error calling first: first() called on vector with no elements
```

**Expected behavior**
Rules are valid and dryRun should succeed

**Version**
```
$ ./vmalert --version
vmalert-20201224-155400-pmm-6401-v1.50.2
```

**Comment by fmbiete** [comment]
One interesting thing is that if we use "with" there is no parse error

Fails to parse
```
dynamic: '{{ $x := query "test_record" | first | value }}{{ if eq 1.0 $x }}one{{ else }}unknown{{ end }}'
```

Succeeds
```
dynamic: '{{ with query "test_record" }}{{ $x := . | first | value }}{{ if eq 1.0 $x }}one{{ else }}unknown{{ end }}{{ end }}'
```

**Comment by f41gh7** [comment]
@hagen1778 can you look?

**Comment by f41gh7** [comment]
Seems like prometheus doesn't validate content of query expression https://github.com/prometheus/prometheus/blob/master/pkg/rulefmt/rulefmt.go#L239
 
 And with label:
 ```dynamic: '{{ $x := query "fsfas12rasup{fas" | first | value }}{{ if eq 1.0 $x }}one{{ else }}unknown{{ end }}' ```
You get  alertling label:
```dynamic="<error expanding template: error executing template __alert_test_alert: template: __alert_test_alert:1:89: executing "__alert_test_alert" at <query "fsfas12rasup{fas">: error calling query: 1:17: parse error: unexpected end of input inside braces>"```

**Comment by valyala** [comment]
The issue should be fixed in the commit 2e2e4f7e21a426fb42b20cc2932e61f533e9515d . It will be included in the next release of `vmagent`.

**Comment by valyala** [comment]
The bugfix has been included in [v1.52.0 release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). @fmbiete , could you  verify whether `vmalert` v1.52.0 works as intended?

**Comment by fmbiete** [comment]
@valyala works for me.
Thank you!
